kotlin/browse-ogc-api-feature-service/src/main/java/com/esri/arcgisruntime/sample/browseogcapifeatureservice/MainActivity.kt

```diff
@@ -0,0 +1,246 @@
+    }
+
+    private val layerFAB: FloatingActionButton by lazy {
+        activityMainBinding.layerFAB
+    }
+
+    private val featureCollectionTitleListView: ListView by lazy {
+        activityMainBinding.layerList
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // authentication with an API key or named user is required to access basemaps and other
+        // location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // initialize the text box with a service url
+        serviceEditText.setText(serviceUrl)
+        // create a map with topographic basemap
+        mapView.map = ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC)
+
+        // load the service
+        loadService()
+
+        setupUI()
+    }
+
+    /**
+     * Loads a new OGC Feature Service with the url in the edit text box. Use the service's info to
+     * get titles for all feature collections. Add those titles to the list view and on tap call
+     * loadLayer.
+     */
+    private fun loadService() {
+
+        // create the OGC API - Features service using the landing URL
+        val service = OgcFeatureService(serviceEditText.text.toString())
+
+        // load the OAFeat service
+        service.loadAsync()
+        service.addDoneLoadingListener {
+
+            // get the service metadata
+            val serviceInfo = service.serviceInfo
+
+            // get a list of available collections
+            val featureCollectionInfos = serviceInfo.featureCollectionInfos
+
+            // get a list of the collection titles
+            val featureCollectionTitles = featureCollectionInfos.map { it.title }
+
+            // create an adapter to show the feature collection titles
+            val featureCollectionTitleAdapter = ArrayAdapter(
+                this,
+                R.layout.feature_collection_title_list_item,
+                featureCollectionTitles
+            )
+
+            featureCollectionTitleListView.apply {
+                // add the adapter to the list view
+                adapter = featureCollectionTitleAdapter
+                // set an on item click listener for items in the list view
+                setOnItemClickListener { _, _, position, _ ->
+                    // load the selected collection
+                    loadLayer(featureCollectionInfos[position])
+                }
+            }
+
+            // expand the layers list
+            layerFAB.isExpanded = true
+        }
+    }
+
+    /**
+     * Load  and query features from the given OGC Feature Collection. Create a Feature Layer from
+     * the table and add it to the map's operational layers.
+     */
+    private fun loadLayer(selectedCollectionInfo: OgcFeatureCollectionInfo) {
+        // create the OGC feature collection table
+        val table = OgcFeatureCollectionTable(selectedCollectionInfo)
```

Comment:
variable name suggestion for readability and context later in this method: 
```suggestion
        val ogcFeatureCollectiontable = OgcFeatureCollectionTable(selectedCollectionInfo)
```